ci/travis/detect-changes.py

```diff
@@ -0,0 +1,26 @@
+# Script used for checking changes for incremental testing cases
+
+import os
+import sys
+
```

Comment:
I think the variables we want to produce are 
- `RAY_TUNE_CHANGES_ONLY`
- `RAY_RLLIB_CHANGES_ONLY`
- `RAY_DOC_CHANGES_ONLY`
- `RAY_JAVA_CHANGES_ONLY`

This means that ALL changes are contained inside of the relevant folders, not SOME changes.

Reply:
Actually the above may not be the best. But anyway, I think we want the following.

- Run **Tune** tests if there are changes to Tune or the Ray Python/C++ backend, but not if the changes are restricted to RLlib/Java/Documentation.
- Run **RLlib** tests if there are changes to Tune/RLlib or the Ray Python/C++ backend, but not if the changes are restricted to Java/Documentation.
- Run **Python tests** if there are changes to the Ray Python/C++ backend, but not if the changes are restricted to Tune/RLlib/Java/Documentation.
- Run the **Linting** build always.
- Run the **Java** test if there are changes to the Ray C++/Java backend, but not if the changes are restricted to Tune/RLlib/Python/Documentation.
- Build the **Linux wheels** if there are changes to Tune/RLlib or the Python/C++ backend but not if the changes are restricted to Java/Documentation.
- Build the **MacOS wheels** if there are changes to Tune/RLlib or the Python/C++ backend but not if the changes are restricted.

Reply:
Let's put all of the logic into this Python script, and simply have it export all relevant variables.

Reply:
Another note is that on the **master** branch, we always need to build the Linux and MacOS wheels.